src/components/Chat.astro

```diff
@@ -54,16 +73,55 @@ import Button from "../components/Button.astro";
     const assistantMessage =
       chatCompletion.choices[0]?.message?.content || "No response";
     chatHistory.push({ role: "assistant", content: assistantMessage });
-
     return assistantMessage;
   }
 
+  // Function to display messaging history
+  function displayMessages() {
+    const historyDiv = document.getElementById("history");
+    historyDiv.innerHTML = ""; // Clear previous messages
+    for (let i = 1; i < chatHistory.length; i++) {
```

Comment:
I usually lean towards `forEach` just because it is easier to read, but `for` work just as well. Here are the docs on [forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#description)

If you did want to change it it would something like
```suggestion
    chatHistory.forEach((i) => {
```
Then just add `);` to the closing for `}`.